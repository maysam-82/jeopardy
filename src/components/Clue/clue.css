.answer-hidden {
	visibility: hidden;
}

.answer-visible {
	visibility: visible;
	animation-name: fade-in;
	animation-duration: 1s;
}

@keyframes fade-in {
	from {
		opacity: 0;
	}
	to {
		opacity: 1;
	}
}
