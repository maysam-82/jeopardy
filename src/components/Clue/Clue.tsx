import React from 'react';
import { ListGroup } from 'react-bootstrap';
import './clue.css';

interface IClueProps {
	answer: string;
	value: number;
	question: string;
}

interface IClueState {
	isAnswerVisible: boolean;
}

class Clue extends React.Component<IClueProps, IClueState> {
	constructor(props: IClueProps) {
		super(props);

		this.state = {
			isAnswerVisible: false,
		};
	}

	onQuestionClickedHandler = () => {
		this.setState((prevState) => ({
			isAnswerVisible: !prevState.isAnswerVisible,
		}));
	};

	render() {
		const { answer, question, value } = this.props;
		const { isAnswerVisible } = this.state;
		return (
			<ListGroup className="mb-2">
				<ListGroup.Item className="font-weight-bold" variant="secondary">
					{value}
				</ListGroup.Item>
				<ListGroup.Item
					action
					variant="warning"
					onClick={this.onQuestionClickedHandler}
				>
					{question}
				</ListGroup.Item>
				<ListGroup.Item
					className={`${isAnswerVisible ? 'answer-visible' : 'answer-hidden'}`}
					variant="success"
				>
					{answer}
				</ListGroup.Item>
			</ListGroup>
		);
	}
}

export default Clue;
