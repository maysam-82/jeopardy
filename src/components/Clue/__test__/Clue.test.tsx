import React from 'react';
import { shallow } from 'enzyme';
import Clue, { IClueProps, IClueState } from '../Clue';

const testClueProps: IClueProps = {
	answer: 'test answer',
	question: 'test question',
	value: 100,
};

describe('Clue component', () => {
	const component = shallow<Clue, IClueProps, IClueState>(
		<Clue {...testClueProps} />
	);
	it('should render Clue component', () => {
		expect(component).toMatchSnapshot();
	});
	it('should have `answer` with value of `test answer`', () => {
		expect(component.find('ListGroupItem').at(2).text()).toEqual(
			testClueProps.answer
		);
	});
	it('should have `question` with value of `test question`', () => {
		expect(component.find('ListGroupItem').at(1).text()).toEqual(
			testClueProps.question
		);
	});
	it('should have `value` with value of `100`', () => {
		expect(component.find('ListGroupItem').at(0).text()).toEqual(
			testClueProps.value.toString()
		);
	});
	it('should have `ListGroupItem` for answer with class name of `answer-hidden`', () => {
		expect(
			component.find('ListGroupItem').at(2).find('.answer-hidden').exists()
		).toBeTruthy();
	});
	describe('When user click on question', () => {
		beforeEach(() => {
			component.find('ListGroupItem').at(1).simulate('click');
		});
		it('should add `answer-visible` class to `ListGroupItem` for answer', () => {
			expect(
				component.find('ListGroupItem').at(2).find('.answer-visible').exists()
			).toBeTruthy();
		});
	});
});
