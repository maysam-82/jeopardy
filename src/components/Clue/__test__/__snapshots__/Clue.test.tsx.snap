// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clue component should render Clue component 1`] = `
<ListGroup
  className="mb-2"
  horizontal={null}
  variant={null}
>
  <ListGroupItem
    active={false}
    className="font-weight-bold"
    disabled={false}
    variant="secondary"
  >
    100
  </ListGroupItem>
  <ListGroupItem
    action={true}
    active={false}
    disabled={false}
    onClick={[Function]}
    variant="warning"
  >
    test question
  </ListGroupItem>
  <ListGroupItem
    active={false}
    className="answer-hidden"
    disabled={false}
    variant="success"
  >
    test answer
  </ListGroupItem>
</ListGroup>
`;
